operator/src/main/resources/application.properties

```diff
@@ -11,6 +11,9 @@ operator.keycloak.image-pull-policy=Always
 quarkus.kubernetes.env.vars.operator-keycloak-image=${operator.keycloak.image}
 quarkus.openshift.env.vars.operator-keycloak-image=${operator.keycloak.image}
 
+quarkus.kubernetes.env.fields."quarkus.operator-sdk.namespaces"=metadata.annotations['olm.targetNamespaces']
+quarkus.openshift.env.fields."quarkus.operator-sdk.namespaces"=metadata.annotations['olm.targetNamespaces']
```

Comment:
I'd like to propose an alternative way as I don't think we should be expecting OLM specific annotations by default. Feels a bit weird, IMHO. :)

Also setting `-Dquarkus.kubernetes.annotations.\"olm.targetNamespaces\"=JOSDK_WATCH_CURRENT` sets annotations on all generated resources (svc accounts, services, ...). I think it's not ideal.

We could try this instead:
* Set `quarkus.operator-sdk.namespaces=JOSDK_WATCH_CURRENT` in this properties file. This will keep current behavior where the Operator watches only current namespace by default. This is also what [non-OLM installation resources](https://github.com/keycloak/keycloak-k8s-resources/tree/nightly/kubernetes) should use.
* Modify the OLM scripts to override the watched namespaces via env vars, i.e. set `QUARKUS_OPERATOR_SDK_NAMESPACES` to reference the `olm.targetNamespace` annotation in the CSV (and nowhere else). Even if it ends up being set to empty string, it should do the trick (observing all namespaces).

With this approach we'd make sure the default generated K8s resources (which are namespaced in any case – especially in the case of roles) are setting the Operator to watch current namespace without any additional configuration. At the same time, the OLM specific annotations would be used only in CSV.

WDYT?

Reply:
> Set quarkus.operator-sdk.namespaces=JOSDK_WATCH_CURRENT in this properties file. This will keep current behavior where the Operator watches only current namespace by default. 

I think my confusion here was that quarkus.operator-sdk.namespaces=JOSDK_WATCH_CURRENT seemed to be forcing the behavior in the image.  However I just checked again and that's not the case.  So it is possible to use that as a default, but that still isn't great - more below.

> This is also what [non-OLM installation resources](https://github.com/keycloak/keycloak-k8s-resources/tree/nightly/kubernetes) should use.

What are these used for?  It doesn't appear that quarkus.operator-sdk.namespaces alone will influence the creation of RoleBindings instead of ClusterRoleBindings.  So you at least need to include a -Dquarkus.kubernetes.namespace so that the ClusterRoleBinding will be valid.  Alternatively quarkus.operator-sdk.controllers.XXX.namespaces can be used, but they definitely do have an effect on the image - so to emit the same kubernetes.yaml with just RoleBindings would take a separate mvn package run specifying quarkus.operator-sdk.controllers.XXX.namespaces for each controller.

> Modify the OLM scripts to override the watched namespaces via env vars, i.e. set QUARKUS_OPERATOR_SDK_NAMESPACES to reference the olm.targetNamespace annotation in the CSV (and nowhere else). Even if it ends up being set to empty string, it should do the trick (observing all namespaces) as the env vars take precedence over properties file.

In this pr the env property is already handled via the application.properties.  My thought was that eventually it would be nice to move away from using create-olm-bundle.sh - that should cut down on the variance between what we're testing via the local/remote tests vs olm.  

Beyond that we don't want the OLM scripts to directly set anything about the namespaces in the CSV - that's a job for OLM itself based upon the InstallMode/operatorgroup.

Reply:
> It is actually forcing the image behaviour but can be easily overridden via env vars.

By forcing I meant that it could not then be overriden via env vars.

> I'm not sure I fully follow. :) The generated kubernetes.yaml is namespaced installation in any case. The service account is namespaced as well. If I understand correctly, even the ClusterRoleBinding is then specific for that single service account in that namespace. What's the difference than from the [current approach](https://github.com/keycloak/keycloak-k8s-resources/blob/795ad1492ef2b6f31ae779da3069b6d7232f5441/kubernetes/kubernetes.yml#L166-L176) with just RoleBindings?

Before any of the pending upstream changes if you remove the namespaces from the controller annotations but set quarkus.operator-sdk.namespaces=JOSDK_WATCH_CURRENT it will create ClusterRoleBindings in the kubernetes.yml:

```
apiVersion: rbac.authorization.k8s.io/v1
kind: ClusterRoleBinding
metadata:
  name: keycloakrealmimportcontroller-cluster-role-binding
roleRef:
  kind: ClusterRole
  apiGroup: rbac.authorization.k8s.io
  name: keycloakrealmimportcontroller-cluster-role
subjects:
  - kind: ServiceAccount
    name: keycloak-operator
```

Note that the subject lacks a namespace, which makes it invalid.  You have to also specify quarkus.kubernetes.namespace for the generated yaml to be valid.

Reply:
@shawkins I see. I got confused before, sorry. :) We need RoleBindings without any namespace specified then (as it is now). It almost feels like a JOSDK bug to me that `quarkus.operator-sdk.namespaces=JOSDK_WATCH_CURRENT` makes it generate ClusterRoleBindings. WDYT?

Reply:
Ah yes, it was fixed by https://github.com/quarkiverse/quarkus-operator-sdk/pull/654.

Then I'd suggest waiting for that fix before proceeding with this PR.

Reply:
Yes https://github.com/quarkiverse/quarkus-operator-sdk/issues/655 and https://github.com/quarkiverse/quarkus-operator-sdk/issues/653 were captured to address JOSDK issues.

Here's a recap of where things are currently:
- removing the namespaces from the controllers causes the kubernetes.yml to use ClusterRoleBindings.
  - Using quarkus.operator-sdk.namespaces=JOSDK_WATCH_CURRENT - does not change the generation of ClusterRoleBindings.
  - Using quarkus.operator-sdk.controllers.xxx.namespaces also does not change the generated kubernetes.yml - it would also need to be expressed as ${quarkus.operator-sdk.namespaces} otherwise it would prevent you from using the olm annotations to drive the namespaces.  
  - Using quarkus.operator-sdk.<controller name>.generate-with-watched-namespaces - does affect the kubernetes.yml, but would also need to be expressed as ${quarkus.operator-sdk.namespaces} otherwise it would prevent you from using the olm annotations to drive the namespaces.

I had initially used quarkus.operator-sdk.controllers.xxx.namespaces the other direction - leave the annotation namespaces set on the controllers, then allow these properties to override things.  The sticking point there however is there isn't a simplistic way to differentate between a missing annotation (non-olm installation) and an empty annotation (all namespaces case), so you have to leave the env field handling to the olm script and not the application.properties.  

I'll revert this pr to something that looks like that, but based upon the generate-with-watched-namespaces.